src/rgw/rgw_formats.cc

```diff
@@ -287,7 +287,9 @@ void RGWFormatter_Plain::dump_value_int(const char *name, const char *fmt, ...)
 class HTMLHelper : public XMLFormatter {
 public:
   static std::string escape(const std::string& unescaped_str) {
-    return escape_xml_str(unescaped_str.c_str());
+    std::stringstream ss;
+    ss << xml_stream_escaper(unescaped_str);
+    return ss.str();
```

Comment:
a few of the callers of `HTMLHelper::escape()` are using streams, so you can use `xml_stream_escaper` directly in those cases. the rest are being passed into `boost::format`, so we do need them as strings - so lets use the same approach here with `std::string::data()` as you did for `escape_str()` in rgw_cls.cc